src/wpantund/NCPInstanceBase-Addresses.cpp

```diff
@@ -30,122 +30,529 @@
+				is_filtered() ? "yes" : "no");
+		} else {
+			snprintf(c_string, sizeof(c_string),
+				"\"%s/%d\", origin:%s, filtered:%s, valid:%u, preferred:%u",
+				in6_addr_to_string(address).c_str(), get_prefix_len(), get_origin_as_string().c_str(),
+				is_filtered() ? "yes" : "no", mValidLifetime, mPreferredLifetime);
+		}
+	}
+
+	return std::string(c_string);
+}
+
+std::string
+NCPInstanceBase::MulticastAddressEntry::get_description(const struct in6_addr &address, bool align) const
+{
+	char c_string[300];
+
+	if (align) {
+		snprintf(c_string, sizeof(c_string), "%-40s origin:%s", in6_addr_to_string(address).c_str(),
+			get_origin_as_string().c_str());
+	} else {
+		snprintf(c_string, sizeof(c_string), "\"%s\", origin:%s", in6_addr_to_string(address).c_str(),
+			get_origin_as_string().c_str());
+	}
+
+	return std::string(c_string);
+}
+
+std::string
+NCPInstanceBase::on_mesh_prefix_flags_to_string(uint8_t flags, bool detailed)
+{
+	char c_string[300];
+
+	if (detailed) {
+		uint8_t preferece = flags & OnMeshPrefixEntry::kPreferenceMask;
+		const char *prio = "none";
+
+		switch (preferece) {
+		case OnMeshPrefixEntry::kPreferenceHigh:
+			prio = "high";
+			break;
+
+		case OnMeshPrefixEntry::kPreferenceLow:
+			prio = "low";
+			break;
+
+		case OnMeshPrefixEntry::kPreferenceMedium:
+			prio = "med";
+			break;
+		}
+
+		snprintf(c_string, sizeof(c_string), "flags:0x%02x [on-mesh:%s def-route:%s config:%s dhcp:%s slaac:%s pref:%s prio:%s]",
+			flags,
+			(flags & OnMeshPrefixEntry::kFlagOnMesh) ? "1" : "0",
+			(flags & OnMeshPrefixEntry::kFlagDefaultRoute) ? "1" : "0",
+			(flags & OnMeshPrefixEntry::kFlagConfigure) ? "1" : "0",
+			(flags & OnMeshPrefixEntry::kFlagDHCP) ? "1" : "0",
+			(flags & OnMeshPrefixEntry::kFlagSLAAC) ? "1" : "0",
+			(flags & OnMeshPrefixEntry::kFlagPreferred) ? "1" : "0",
+			prio
+		);
+	} else {
+		snprintf(c_string, sizeof(c_string), "%s(0x%02x)", flags_to_string(flags, "ppPSDCRM").c_str(), flags);
+	}
+
+	return c_string;
+}
+
+std::string
+NCPInstanceBase::OnMeshPrefixEntry::get_description(const struct in6_addr &address, bool align) const
+{
+	char c_string[300];
+
+	if (align) {
+		snprintf(c_string, sizeof(c_string), "%-22s prefix_len:%-4d origin:%-8s stable:%s %s",
+			in6_addr_to_string(address).c_str(), get_prefix_len(), get_origin_as_string().c_str(),
+			is_stable() ? "yes" : "no ", on_mesh_prefix_flags_to_string(get_flags(), true).c_str());
+
+	} else {
+		snprintf(c_string, sizeof(c_string), "\"%s/%d\", origin:%s, stable:%s, flags:%s, ",
+			in6_addr_to_string(address).c_str(), get_prefix_len(), get_origin_as_string().c_str(),
+			is_stable() ? "yes" : "no", on_mesh_prefix_flags_to_string(get_flags()).c_str());
+	}
+
+	return std::string(c_string);
+}
+
+// ========================================================================
+// MARK: Global Entries Management
 
+bool
+NCPInstanceBase::should_filter_address(const struct in6_addr &address, uint8_t prefix_len)
```

Comment:
The method `should_filter_address` basically let's sub-classes to filter certain addresses. I added a simple implementation in base class to avoid making it an abstract virtual function (if plugin sub-class want to avoid this). 

Spinel plugin uses this method to filter RLOC based addresses. We can consider having the addresses filtered by NCP or by the sub-class directly but I felt having the full list of NCP added addresses and showing which ones were filtered can be helpful for debugging.